docs: Update README

deekerno opened the pull request:
Closes #393.

## Changelog
- Update `README`

## Notes
I decided to update the README into more of a "quickstart" style as we've gotten `forc index` to a point at which we can endorse it as the recommended usage method for newer users. The intention is to use a very simple example to lower the barrier to entry such that people are not put off by potential difficulties in learning to use the indexer. There are numerous links to our book and other Fuel components so that users that want to go deeper can easily find information.

An update to the book will follow this PR as it needs to be updated with information about `forc index` among other things.

ra0x3 reviewed (changes requested):
Looks mostly good so far. A few notes:

- Can we use markdown anchor links at the top for each section (sections being fuelup, forc-index, databases, quickstart, etc) so that users can click down to the section they want without having to scroll? Also acts as a little way to chunk the README into digestable parts
- This is gonna be a PITA, and relates to the work I'm doing in https://github.com/FuelLabs/fuel-indexer/pull/409 but we're gonna have to explain how to install the database backends for sqlite + postgres on each platform (just darwin, linux for now maybe?).
  - Again, my intention is to add this info to https://github.com/FuelLabs/fuel-indexer/pull/409 so feel free to co-author that PR
  - For Mac, we'll have to explain that they need to download homebrew and `brew install postgresql` (amongst other things)
  - For Linux, we'll have to follow whatever Linux docs.
  - In https://github.com/FuelLabs/fuel-indexer/pull/409 I'm basically copy/pasting other docs into our docs (so I don't have to write them myself, yet they're very detailed)
    - [Checkout the new doc for sqlx I added](https://github.com/FuelLabs/fuel-indexer/blob/517583df0276fbf9cebee14fae0a5815927ebff9/docs/src/getting-started/application-dependencies/sqlx.md)
    - One of the biggest pieces of feedback we've  gotten so far is "comprehensive" documentation - not assuming that users have the know-how or the dependencies already installed

That's it for now. But looking good

deekerno commented:
@ra0x3 Addressed your feedback. I also added a bit about Rust as if we're trying to avoid assumptions, we probably shouldn't assume that they have Rust installed either.
